### PR TITLE
[IMP] im_livechat: introduce PublicLivechatGlobal/welcomeMessages

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2963063

### CODE REVIEW

File: addons/im_livechat/static/src/models/public_livechat_global.js [hunk 1]
```diff
@@ -83,6 +83,15 @@ registerModel({
             }
             return clear();
         },
+        _computeWelcomeMessages() {
+            const welcomeMessages = this.messaging.publicLivechatGlobal.messages.filter((message) => {
```
Comment (seb-odoo) [inline addons/im_livechat/static/src/models/public_livechat_global.js:87]: this.messaging.publicLivechatGlobal = this

Comment (alexkuhn) [reply addons/im_livechat/static/src/models/public_livechat_global.js]: change done

File: addons/im_livechat/static/src/models/public_livechat_global.js [hunk 2]
```diff
@@ -83,6 +83,15 @@ registerModel({
             }
             return clear();
         },
+        _computeWelcomeMessages() {
+            const welcomeMessages = this.messages.filter((message) => {
+                return message.id && typeof message.id === 'string' && message.id.startsWith('_welcome_');
+            });
+            if (welcomeMessages.length === 0) {
+                return clear();
+            }
```
Comment (seb-odoo) [inline addons/im_livechat/static/src/models/public_livechat_global.js:92]: this is not useful, many relations accept replace empty array

**Review by seb-odoo — APPROVED** [review]

File: addons/im_livechat/static/src/models/public_livechat_global.js (hunk 2 above)
Comment (alexkuhn) [reply addons/im_livechat/static/src/models/public_livechat_global.js]: change done

**Comment by alexkuhn** [comment]
Merged in grouped PR: https://github.com/odoo/odoo/pull/98987

**Pull request closed without merging**